annotator/__main__.py

```diff
@@ -0,0 +1,440 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+from future.utils import iteritems
+from six.moves.urllib.parse import urlparse
+from six.moves.urllib.request import urlopen
+import subprocess
+import os
+import sys
+import json
+import six
+import argparse
+import getpass
+import pandas
+import synapseutils
+import synapseclient
+from synapseclient import Entity, Project, Column, Team, Wiki, Folder
+from annotator import schema
+
+
+def synapseLogin():
+    """
+    First tries to login to synapse by finding the local auth key cached on user's computing platform, if not found,
+    prompts the user to provide their synapse user name and password, then caches the auth key on their computing
+    platform.
+
+    :return:
+    """
+    try:
+        syn = synapseclient.login()
+    except Exception as e:
+        print('Please provide your synapse username/email and password (You will only be prompted once)')
+        username = input("Username: ")
+        password = getpass.getpass(("Password for " + username + ": ").encode('utf-8'))
+        syn = synapseclient.login(email=username, password=password, rememberMe=True)
+
+    return syn
+
+
+def updateTable(syn, tableSynId, newTable, releaseVersion):
+    """
+    Gets the current annotation table, deletes all it's rows, then updates the table with new content generated
+    from all the json files on synapseAnnotations. In the process also updates the table annotation to the latest
+    releaseversion.
+
+    :param syn:
+    :param tableSynId:
+    :param newTable:
+    :param releaseVersion:
+    :return:
+    """
+    currentTable = syn.tableQuery("SELECT * FROM %s" % tableSynId)
+
+    # If current table has rows, delete all the rows
+    if currentTable.asRowSet().rows:
+        deletedRows = syn.delete(currentTable.asRowSet())
```

Comment:
Do we want to delete all annotations? Or just update? Should there be an option to do both?

Reply:
I had an update function, but in this case @kdaily suggested to remove the complexity of updating a table and just delete the entire table. It happens to be faster too :)